datajunction-clients/python/datajunction/admin.py

```diff
@@ -53,26 +58,36 @@ def add_engine(
         self,
         name: str,
         version: str,
-        uri: Optional[str],
-        dialect: Optional[str],
+        uri: Optional[str] = None,
+        dialect: Optional[str] = None,
+        skip_if_exists: bool = False,
     ) -> None:
         """
         Add an engine.
         """
-        response = self._session.post(
-            "/engines/",
-            json={
-                "name": f"{name}",
-                "version": f"{version}",
-                "uri": f"{uri}",
-                "dialect": f"{dialect}",
-            },
-            timeout=self._timeout,
-        )
-        json_response = response.json()
+        params = {
+            "name": f"{name}",
+            "version": f"{version}",
+        }
+        if uri:  # pragma: no cover
+            params["uri"] = f"{uri}"
```

Comment:
minor: this can just be `uri`

Reply:
yes, but only if I change the default value to empty string otherwise we get:
```
>>> a = None
>>> f"{a}"
'None'
```